### PR TITLE
External etcd support for Nutanix

### PR DESCRIPTION
**Pull request by adiantum**
*Description of changes:*
External etcd support for Nutanix
 - change Nutanix provider code
 - change cp-template
 - add new unit tests
 - change unit tests

*Testing (if applicable):*
```
$ eksctl anywhere create cluster -f ./eksa-test-ext-etcd-new-ubuntu2204-1-28.yaml --bundles-override bin/local-bundle-release.yaml
Performing setup and validations
odd number of arguments passed as key-value pairs for logging	{"ignored key": "test-ext-etcd-new"}
ValidateClusterSpec for Nutanix datacenter
Warning: Skipping TLS validation for insecure connection to Nutanix Prism Central; this is not recommended for production use
✅ Nutanix Provider setup is valid
✅ Validate OS is compatible with registry mirror configuration
✅ Validate certificate for registry mirror
✅ Validate authentication for git provider
✅ Validate cluster's eksaVersion matches EKS-A version
Creating new bootstrap cluster
Provider specific pre-capi-install-setup on bootstrap cluster
Installing cluster-api providers on bootstrap cluster
Provider specific post-setup
Creating new workload cluster
Installing networking on workload cluster
Creating EKS-A namespace
Installing cluster-api providers on workload cluster
Installing EKS-A secrets on workload cluster
Installing resources on management cluster
Moving cluster management from bootstrap to workload cluster
Installing EKS-A custom components (CRD and controller) on workload cluster
Installing EKS-D components on workload cluster
Creating EKS-A CRDs instances on workload cluster
Installing GitOps Toolkit on workload cluster
GitOps field not specified, bootstrap flux skipped
Writing cluster config file
Deleting bootstrap cluster
🎉 Cluster created!
--------------------------------------------------------------------------------------
The Amazon EKS Anywhere Curated Packages are only available to customers with the
Amazon EKS Anywhere Enterprise Subscription
--------------------------------------------------------------------------------------
Enabling curated packages on the cluster
Installing helm chart on cluster	{"chart": "eks-anywhere-packages", "version": "0.0.0-3b5440427449629eeb5c981bdd5743d37096c4c5"}

$ kubectl get nodes
NAME                                 STATUS   ROLES           AGE   VERSION
test-ext-etcd-new-fnrfk              Ready    control-plane   39m   v1.28.5-eks-dd170a8
test-ext-etcd-new-md-0-92fnm-tkp88   Ready    <none>          39m   v1.28.5-eks-dd170a8
test-ext-etcd-new-md-0-92fnm-w2kdn   Ready    <none>          40m   v1.28.5-eks-dd170a8
test-ext-etcd-new-t674l              Ready    control-plane   41m   v1.28.5-eks-dd170a8

$ kubectl get machines -n eksa-system
NAME                                 CLUSTER             NODENAME                             PROVIDERID                                       PHASE     AGE   VERSION
test-ext-etcd-new-etcd-dslgk         test-ext-etcd-new                                        nutanix://652b8422-925a-496e-9d3f-ee3fc06d4fe9   Running   38m
test-ext-etcd-new-etcd-kxxcr         test-ext-etcd-new                                        nutanix://8ece6cce-b992-4426-8548-83f935ffeb4d   Running   38m
test-ext-etcd-new-etcd-zw8bj         test-ext-etcd-new                                        nutanix://14b73305-e32f-4611-8d8f-ad19b6195f1c   Running   38m
test-ext-etcd-new-fnrfk              test-ext-etcd-new   test-ext-etcd-new-fnrfk              nutanix://901a25d6-cfee-447f-83bd-33024a8ad637   Running   38m   v1.28.4-eks-1-28-14
test-ext-etcd-new-md-0-92fnm-tkp88   test-ext-etcd-new   test-ext-etcd-new-md-0-92fnm-tkp88   nutanix://2f5dc3b7-c2ba-4100-a0aa-977d4cc7c8b1   Running   38m   v1.28.4-eks-1-28-14
test-ext-etcd-new-md-0-92fnm-w2kdn   test-ext-etcd-new   test-ext-etcd-new-md-0-92fnm-w2kdn   nutanix://63b9359f-2e7f-4366-957a-99340e4381f6   Running   38m   v1.28.4-eks-1-28-14
test-ext-etcd-new-t674l              test-ext-etcd-new   test-ext-etcd-new-t674l              nutanix://c5b857c5-466d-445c-af7e-cc6265e3772c   Running   38m   v1.28.4-eks-1-28-14
```
Screenshot 2024-02-12 at 13 43 44

*Documentation added/planned (if applicable):*
External etcd configuration will be added to Nutanix provider docs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/nutanix/config/cp-template.yaml [hunk 1]
```diff
@@ -389,6 +396,118 @@ spec:
...
+{{- if .etcdNtpServers }}
+    ntp:
+      enabled: true
+      servers: {{ range .etcdNtpServers }}
+      - {{ . }}
+      {{- end }}
+{{- end }}
+    cloudInitConfig:
+      version: {{.externalEtcdVersion}}
+      installDir: "/usr/bin"
```
Comment (abhinavmpandey08) [inline pkg/providers/nutanix/config/cp-template.yaml:420]: Can you add `etcdReleaseURL` here similar to this https://github.com/aws/eks-anywhere/blob/main/pkg/providers/vsphere/config/template-cp.yaml#L562-L564

Comment (adiantum) [reply pkg/providers/nutanix/config/cp-template.yaml:420]: Fixed

**Pull request merged**